<!-- -->
<launch>
  <arg name='host' default='localhost'/>
  <arg name='port' default='2000'/>
  <arg name='timeout' default='10'/>
  <arg name="role_name" default="ego_vehicle" />

  <!-- <node pkg="global_planner" type="global_path_publisher.py" name="global_path_publisher" output="screen">
    <param name="role_name" value="$(arg role_name)" />
    <param name="host" value="$(arg host)" />
    <param name="port" value="$(arg port)" />
    <param name="timeout" value="$(arg timeout)" />
  </node> -->

  <!-- <arg name="role_name" default="ego_vehicle"/> -->
  <arg name="avoid_risk" default="True"/>
  <arg name="control_time_step" default="0.05"/>

  <node pkg="carla_ad_agent" type="ad_agent.py" name="carla_ad_agent_$(arg role_name)" output="screen">
    <param name="role_name" value="$(arg role_name)" />
    <param name="avoid_risk" value="$(arg avoid_risk)" />
  </node>

  <node pkg="mpc_controller" type="mpc_controller.py" name="mpc_controller_$(arg role_name)" output="screen">
    <param name="role_name" value="$(arg role_name)" />
    <param name="control_time_step" value="$(arg control_time_step)" />
  </node>

</launch>
